src/decorateStatus.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2019 Google LLC. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as extend from 'extend';
+
+/**
+ * @const {object} - A map of protobuf codes to HTTP status codes.
+ * @private
+ */
+const GRPC_ERROR_CODE_TO_HTTP = [
```

Comment:
I hate this array so much. It's a plain mapping from gRPC error codes to HTTP error codes which does not make much sense. `@grpc/grpc-js` has its own mapping from error codes to text descriptions:

```
> require('@grpc/grpc-js').status
{
  '0': 'OK',
  '1': 'CANCELLED',
  '2': 'UNKNOWN',
  '3': 'INVALID_ARGUMENT',
  '4': 'DEADLINE_EXCEEDED',
  '5': 'NOT_FOUND',
  '6': 'ALREADY_EXISTS',
  '7': 'PERMISSION_DENIED',
  '8': 'RESOURCE_EXHAUSTED',
  '9': 'FAILED_PRECONDITION',
  '10': 'ABORTED',
  '11': 'OUT_OF_RANGE',
  '12': 'UNIMPLEMENTED',
  '13': 'INTERNAL',
  '14': 'UNAVAILABLE',
  '15': 'DATA_LOSS',
  '16': 'UNAUTHENTICATED',
  OK: 0,
  CANCELLED: 1,
  UNKNOWN: 2,
  INVALID_ARGUMENT: 3,
  DEADLINE_EXCEEDED: 4,
  NOT_FOUND: 5,
  ALREADY_EXISTS: 6,
  PERMISSION_DENIED: 7,
  RESOURCE_EXHAUSTED: 8,
  FAILED_PRECONDITION: 9,
  ABORTED: 10,
  OUT_OF_RANGE: 11,
  UNIMPLEMENTED: 12,
  INTERNAL: 13,
  UNAVAILABLE: 14,
  DATA_LOSS: 15,
  UNAUTHENTICATED: 16
}
```

and I feel it's much better to just provide this explanation rather than try to map it to HTTP code (e.g. `15 DATA_LOSS` does not really translate to `500 Internal Server Error`, and `11 OUT_OF_RANGE` cannot be replaced with `400 Bad Request` since it loses important details.

So, let's get rid of this mapping. If it makes it to be a semver major change, I'm totally fine with it.

Reply:
While I agree we should change this, I kinda don't want to touch too much here at once.  Happy to create a cleanup item to swing back around.